src/librustc/session/config.rs

```diff
@@ -1663,7 +1665,24 @@ pub fn build_session_options_and_crate_config(matches: &getopts::Matches)
         early_error(error_format, "Value for codegen units must be a positive nonzero integer");
     }
 
-    if cg.lto && debugging_opts.incremental.is_some() {
+    let incremental = match (&debugging_opts.incremental, &cg.incremental) {
+        (&Some(ref path1), &Some(ref path2)) => {
+            if path1 == path2 {
```

Comment:
Shouldnt this be a `!=`, currently it errors when there is no conflict

Reply:
Good catch! Thanks, @bjorn3.